day3b.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+from dataclasses import dataclass
+import re
+from collections import defaultdict
+
+NUMBER_PATERN = re.compile(r"\d+")
+
+with open("input3a.txt", "r") as fp:
+    lines = fp.read().strip().splitlines()
+
+@dataclass
+class PartNumber:
+    row: int
+    start: int
+    end: int
+    value: int
+
+
+def find_maybe_part_numbers(lines: list[str]) -> list[PartNumber]:
```

Comment:
A thought about how to make this code more readable:

1. Rename `find_maybe_part_numbers` -> `find_part_numbers`.
2. Do what @zianwar was suggesting -- combine `get_neighbor_positions` and `has_symbol` into a single function called `is_valid`.
3. Call `is_valid` at line 23, passing the part number.